### DISCUSSION TITLE
Default Value for Many-to-One Dropdown using $CURRENT_USER

### DISCUSSION THREAD

**Discussion by tinhXEM** [directus/directus#22446] [Feature Requests]
### Summary

This feature request aims to provide a default value for many-to-one dropdown fields using $CURRENT_USER. When creating a new item, the dropdown should automatically pre-select the current user's ID if applicable.

### Basic Example

<img width="890" alt="image" src="https://github.com/directus/directus/assets/127169920/7f0761bd-efcf-4b52-973a-26b0d9789442">

### Motivation

Currently, Directus does not have a built-in way to automatically pre-select the current user's ID in many-to-one dropdown fields when creating a new item. This feature would save time and reduce errors in user selection, especially for collections where items are typically associated with the creator.

### Detailed Design

The $CURRENT_USER variable should be supported as a default value for many-to-one dropdown fields. When a new item is created, the dropdown field will automatically select the current user's ID if the field is configured to use $CURRENT_USER as its default value.

### Must Have:
- Support for $CURRENT_USER as a default value in many-to-one dropdown fields (and others type just save id as string)
- Automatic pre-selection of the current user's ID when creating a new item.

### Should Have:
- Fallback to an empty selection or another default value if $CURRENT_USER cannot be determined.

### Could Have:
- Compatibility with other role-based special variables.

### Drawbacks

N/A

### Alternatives

N/A

### Adoption Strategy

N/A

**Comment by onigor** [comment]
+1

**Comment by xndyz** [comment]
This seems like a common usecase for authoring posts for example, surprised it hasn't been implemented

**Reply by rijkvanzanten** [reply]
The ability to set the current user ID on save already exists by setting the "On Create" option under Field > Schema to "Save Current User ID". This feature requests requests it as a default value though, which behaves slightly differently. The save current user id feature will set the current user id on save no matter what, whereas a default value can be changed at will during creation

**Comment by burka** [comment]
It would be great if that would work for `$CURRENT_POLICIES` or `$CURRENT_ROLE` or `$CURRENT_ROLES` too.

We use that to set permissions per item. Currently we set the Role in a hook if missing, but it would be great if we could set the default value and have a better user experience and make it easier to select more than the default role.

**Comment by carwei** [comment]
The $CURRENT_USER in default values would be really useful.

**Comment by pipozzz** [comment]
It is possible to set this option On create, after creationg of this field somehow ?

**Reply by rijkvanzanten** [reply]
![CleanShot 2025-04-07 at 11 42 27@2x](https://github.com/user-attachments/assets/5c6810b3-1bf6-48b8-94a3-1114699b8a7d)

Yeah it's under the Schema tab 👍🏻
